shenanigan/utils/logger.py

```diff
@@ -14,18 +12,32 @@ def __init__(self, path, use_pretrained=False):
         self.path = path
         self.use_pretrained = use_pretrained
         self.columns = None
+        self.previous_epoch = self._set_previous_epoch()
 
     def __call__(self, metrics_dict):
         """ Log metrics to file """
+        assert self.previous_epoch != metrics_dict['epoch'], \
+            'The new epoch should not be the same as the previous epoch'
```

Comment:
I ran some tests and this didn't seem to happen though. The epoch counter after using a pretrained model looked like it started where it left off. I might be misunderstanding when this could potentially be a problem maybe?

Reply:
Chatted offline - Will make changes today